lib/ClangImporter/ImportDecl.cpp

```diff
@@ -9735,7 +9735,7 @@ static void loadAllMembersOfRecordDecl(StructDecl *recordDecl) {
       continue;
 
     for (auto found : recordDecl->lookupDirect(name)) {
-      if (addedMembers.insert(found).second)
+      if (addedMembers.insert(found).second && found->getDeclContext() == recordDecl)
```

Comment:
Hmm. I'm not sure this is the right place for this type of check. I feel like this might be pointing to an incorrect predicate in the `copy_if` in `ClangDirectLookupRequest::evaluate`.

Reply:
I was trying to follow your clue about the `copy_if` and the `ClangDirectLookupRequest::evaluate`, and I might had get lost a little, but I found that in https://github.com/apple/swift/blob/d08ecf359d1204f8bda8e94f16ff5cc3cb1f4991/lib/ClangImporter/ClangImporter.cpp#L4148-L4155 `ClangRecordMemberLookup` seems to be doing a similar filtering already (if I am not mistaken). However, in that case, there's a `getClangContext` on the right side, and the returned value is derived from `importDeclDirectly`, so it might not be the same filtering at all.

Did you find out something else?

Reply:
I stumbled into this same issue and put up https://github.com/apple/swift/pull/40354. We're going to have this filtering every place where we use name lookup (which looks across all extensions) but only want to filter down to the members in a given context.  I don't think we need any refactoring here; the "are we adding to the right context" check is appropriate and straightforward.

Reply:
Thanks for that PR. Glad that you found the same solution. I will close this PR. The test is slightly different, but I think both tests will end up testing the same thing.